### PR TITLE
Layers View: Fix Union functionality and add Intersection functionality

### PR DESCRIPTION
**Pull request by OrionsGuardian**
### Prerequisites

- [ ] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

The Layers View now has the option of choosing between Union mode and Intersection mode.

The functionality to perform a Union between selected layers has been corrected.
New functionality to perform an Intersection has now been implemented.

### Alternate Designs

It may be possible to use bit masking and bit shifting to achieve the same result,
but it adds unnecessary complexity.

### Why Should This Be In Core?

Union and Intersection are standard functions to perform between data sets.

### Benefits

Improved functionality.

### Verification Process

Create a new graph
Open the Layers View
Add layers to reach a total of 4 layers.
* Select all 4 layers.
* * Add a node (Vertex#0) to the graph ... note: it will be assigned a layer_mask representing the 4 selected layers.
* Select only layers 1 and 4.
* * Add a node (Vertex#1) to the graph ... this will be assigned to layers 1 and 4.
* Select only layers 1 and 2.
* * Add a node (Vertex#2) to the graph ... this will be assigned to layers 1 and 2.
* Select only layers 2 and 3.
* * Add a node (Vertex#3) to the graph ... this will be assigned to layers 2 and 3.
* Select only layers 3 and 4.
* * Add a node (Vertex#4) to the graph ... this will be assigned to layers 3 and 4.

TEST with UNION mode selected ...
* Select only layer 1:
* * Should only see Vertices V#0, V#1, V#2
* Select only layer 4:
* * Should only see Vertices V#0, V#1, V#4
* Select only layers 2 and 4:
* * Should see **all** Vertices V#0, V#1, V#2, V#3, V#4
* Select layers 1 and 2:
* * Should see Vertices V#0, V#1, V#2, V#3
* Select only layers 3 and 4:
* * Should see Vertices V#0, V#1, V#3, V#4

TEST with INTERSECTION mode selected ...
* Select only layer 1:
* * Should see Vertices V#0, V#1, V#2
* Select only layer 4:
* * Should see Vertices V#0, V#1, V#4
* Select only layers 2 and 4:
* * Should **only** see Vertex V#0
* Select layers 1 and 2:
* * Should **only** see Vertices V#0, V#2
* Select only layers 3 and 4:
* * Should **only** see Vertices V#0, V#4

Also need to test with queries defined in the layer.
Open/Expand the Layer 3 segment.
Add the Vertex Query: `Label contains "1"`
This will make Vertex#1 be treated as though it also belongs to layer 3.
So it will now be included in the nodes shown if you select only layers 3 and 4.

### Applicable Issues

#2170

### CODE REVIEW

**Review by Delphinus8821 — APPROVED** [review]

**Review by Quasar985 — APPROVED** [review]

**Comment by antares1470** [comment]
Does this fix also address #1582 and/or #809?

File: CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/query/BitMaskQueryCollection.java [hunk 1]
```diff
@@ -164,22 +164,35 @@ public long updateBitMask(long bitMask) {
...
+    
     /**
      * Update the overall bit mask attribute and the attribute for which queries are currently active
      * 
      * @param graph
      * @param bitMaskAttributeId
      * @param visibleAttributeId
      */
-    public void updateBitMasks(final GraphWriteMethods graph, final int bitMaskAttributeId, final int visibleAttributeId) {
+    public void updateBitMasks(final GraphWriteMethods graph, final int bitMaskAttributeId, final int visibleAttributeId, boolean unionMode) {
```
Comment (antares1470) [inline CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/query/BitMaskQueryCollection.java:182]: parameter final

**Review by antares1470 — CHANGES_REQUESTED** [review]
Are you able to update the Layers View help page and add an entry to whatsnew?

**Comment by OrionsGuardian** [comment]
> Does this fix also address #1582 and/or #809?

Yes for both ... except that it does not move Named Selection functionality into the Layers View (nor should it be moved).

**Review by antares1470 — APPROVED** [review]

**Pull request merged**